op-node: change default L1 cache size to 900

sebastianst opened the pull request:
In #13772, the L1 cache size was made configurable and the cap of 1000 was removed. This implied that the old default value (before applying the cap) of 2/3 the sequencing window was now active.

However, high-throughput chains don't need such a high cache size under normal conditions, so memory usage unexpectedly jumped for most node operators.

This PR adds back a lower default value of 900, which is a better default for most nodes. Only low-throughput chains with infrequent safe head increments can now use this flag to set a higher value.

bitwiseguy commented on op-node/flags/flags.go:
```diff
@@ -182,8 +182,12 @@ var (
 		Name: "l1.cache-size",
 		Usage: "Cache size for blocks, receipts and transactions. " +
-			"It's optional and a sane default of 3/2 the sequencing window size is used if this field is set to 0.",
+			"If this field is set to 0, a default of 3/2 the sequencing window size is used. " +
```
Should this say "1/4 the sequencing window"? As far as I can tell the [standard sequencing window is 3600 L1 blocks](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/mainnet/op.toml#L12). So 1/4 * 3600 = 900, which is the default value this pr introduces

sebastianst replied:
It's indeed wrong, it should be "2/3", not "3/2". Note that the default is the value of `Value = 900`. Only if this is set to `0` is `2/3 seq window` chosen.

sebastianst commented on op-node/flags/flags.go:
```diff
@@ -182,8 +182,12 @@ var (
 		Name: "l1.cache-size",
 		Usage: "Cache size for blocks, receipts and transactions. " +
-			"It's optional and a sane default of 3/2 the sequencing window size is used if this field is set to 0.",
+			"If this flag is set to 0, 2/3 of the sequencing window size is used. " +
```
The more I think about it, the more I'm thinking that we could just remove this behavior from the L1Client setup. The cache size is so individual per network, so maybe trying to be smart here is just complicating things.

sebastianst replied:
But then again, there are internal tests that uses the default L1 client constructor, so it helps to have 0 get translated into 2/3 seq window...

sebastianst replied:
will leave this behavior in for now.

bitwiseguy commented on op-node/flags/flags.go:
```diff
@@ -182,8 +182,12 @@ var (
-			"It's optional and a sane default of 3/2 the sequencing window size is used if this field is set to 0.",
+			"If this flag is set to 0, 2/3 of the sequencing window size is used. " +
+			"The default value of 900 (~3h of L1 blocks) is good for (high-throughput) networks that see frequent safe head increments. " +
+			"On (low-throuhgput) networks with infrequent safe head increments, it is recommended to set this value to 0, " +
```
Typo: `low-throuhgput` --> `low-throughput`

bitwiseguy commented on op-node/flags/flags.go:
```diff
@@ -182,8 +182,12 @@ var (
+			"On (low-throuhgput) networks with infrequent safe head increments, it is recommended to set this value to 0, " +
+			"or a value that well covers the typical span between safe head increments.",
 		EnvVars:  prefixEnvVars("L1_CACHE_SIZE"),
+		Value:    900, // ~3h of L1 blocks
```
It seems better/safer to me to default to the low-throughput of 2400 blocks, which also works for high-throughput chains. Then high-throughput chains could adjust that lower as an optimization.

Also a little confusing to me to express this setting in different terms: 2/3 sequencer window instead of an L1 block amount as is used for the "Value". For consistency I'd prefer to say 2400 instead of 2/3 seq window

sebastianst replied:
The purpose of this PR was to _lower_ the default. If we wanted to keep it at 2400, we could just close this PR 😅 

"2/3 seq window" is more general than 2400, because chains can be configured to use different seq window sizes, although this is very uncommon.

Most node ops are actually running nodes for high-throughput chains, so when they upgraded, including us, they/we saw an unexpected rise in memory going over 4GB. So it's more optimal to have a lower default memory consumption in op-geth and only have low-through chain nodes explicitly raise the cache.

sebastianst replied:
added a bit of clarification to the flag description

sebastianst committed:
op-node: change default L1 cache size to 900

In #13772, the L1 cache size was made configurable and the cap of 1000
was removed. This implied that the old default value (before applying
the cap) of 2/3 the sequencing window was now active.

However, high-throughput chains don't need such a high cache size under
normal conditions, so memory usage unexpectedly jumped for most node
operators.

This PR adds back a lower default value of 900, which is a better
default for most nodes. Only low-throughput chains with infrequent safe
head increments can now use this flag to set a higher value.
